### PR TITLE
added allocate memory methods to classpathentries

### PR DESCRIPTION
**Pull request by VinoshanT**
Signed-off-by: Vinoshan Tharmalenkam <Vinoshan.Tharmalenkam@ibm.com>

### CODE REVIEW

**Review by Ershany-IBM — APPROVED** [review]

File: runtime/oti/j9nonbuilder.h [hunk 1]
```diff
@@ -4709,6 +4709,8 @@ typedef struct J9InternalVMFunctions {
 	U_32 ( *getVMRuntimeState)(struct J9JavaVM *vm);
 	BOOLEAN ( *updateVMRuntimeState)(struct J9JavaVM *vm, U_32 newState);
 	U_32 ( *getVMMinIdleWaitTime)(struct J9JavaVM *vm);
+	void* ( *mem_allocate_memory)(uintptr_t byteAmount) ;
```
Comment (tajila) [inline runtime/oti/j9nonbuilder.h:4712]: for clarity I would recommend renaming this to `image_mem_allocate_memory` or something else that distinguishes this from the normal memory allocation.

**Comment by akshayben** [comment]
Once the jvmimageportlib pr goes in we could define mem_alloc_memory where j9mem_allocate_mem.

**Comment by akshayben** [comment]
that way there wouldn't be a need to place the functions inside internalfuncs

File: runtime/util/cphelp.c [hunk 2]
```diff
@@ -207,7 +210,13 @@ addJarToSystemClassLoaderClassPathEntries(J9JavaVM *vm, const char *filename)
 		classPathLength += oldEntries[entryIndex].pathLength + 1;	/* add 1 for a null character */
 	}
 	/* Copy and grow the classPathEntries array */
-	newEntries = (J9ClassPathEntry*) j9mem_allocate_memory(sizeof(J9ClassPathEntry) * (entryCount + 1) + classPathLength, J9MEM_CATEGORY_CLASSES);
+
+	if (IS_COLD_RUN(vm)) {
+		newEntries = (J9ClassPathEntry*) vmFuncs->mem_allocate_memory(sizeof(J9ClassPathEntry) * (entryCount + 1) + classPathLength);
```
Comment (akshayben) [inline runtime/util/cphelp.c:215]: function should take in category also?

**Comment by akshayben** [comment]
Vino we could create a macro called IMAGEPORT_ACCESS_FROM_JAVAVM that follows the same convention as PORT_ACCESS_FROM_JAVAVM and that we we don't need our own mem_allocate but we could just call j9mem_allocate_memory

**Comment by akshayben** [comment]
This PR is replaced by another PR. Will be closed

**Pull request closed without merging**